ADR/partner-tasks.md

```diff
@@ -0,0 +1,34 @@
+# Partner Tasks in Build/Test Pipelines
+
+Date: 2023-03-06
+
+## Status
+
+Provisional
+
+## Context
+
+* As a Red Hat Partner, I would like to offer our service's capability as a Tekton Task that would be executed in a user's build/test Pipeline on StoneSoup.
+* As a StoneSoup maintainer, I would like to provide a way to ensure the above Tekton Task is validated and approved for use in the build/test Pipeline on StoneSoup.
+
+Associated business ask
+https://issues.redhat.com/browse/STONE-549
+
+## Decision
+
+### Plumbing
+
+1. Setup a new Git repository.
```

Comment:
Is it possible to write why we can't use tekton-hub?
https://github.com/tektoncd/hub https://hub.tekton.dev/

Reply:
Is this where our current Tasks come from today?  If so then I do think we should be consistent here.

Reply:
It has always been my understanding that we never wanted to enable "arbitrary" leveraging of tasks from `tekton-hub` because it isn't _necessarily_ trusted task definition. There were some initial noodling of us being able to host our own form of `tekton-hub` containing _trusted_ tasks which have gone through certain vetting. This also enables us to have a buffer from the task HEADs so that we might have time to catch potential breaking changes before they affect the pipeline.

Reply:
@gbenhaim mainly because the hub is deprecated. And no this is not where stonesoup tasks are going today.
Upstream, the "platform" of choice to get tasks indexed and searchable is artifacthub now, and artifacthub can index any number of catalog, including ours (severals if we want), if they follow the correct organizations (versionning, …)

Reply:
Added a section called "Alternatives".